### ISSUE TITLE
Bug: Should not allow `using let = h();`

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#7554]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
{
  using let = h();
}
```

- [TS Playground](https://www.typescriptlang.org/play?ssl=3&ssc=2&pln=1&pc=1#code/N4KABGCuDOCWB2BzMAbApgFzAXjACwAoBKAbhAF8g)
- [Prettier Playground (babel)](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwA6UAEmCuBnASygHNMAbOGTAXkwAsAKASgG4MBfEAGhAgAcYBaHmSgAhgCcJEAO4AFSQhEoxZGWICeIngCMJYsAGtKAZTEBbOABkicZADNVeOLv1HTfA0WLIYEnC4gzuYEvv6BhCQUAIo4EPAOToEAVngAHibeMXEJSI5kzjwAjjlwctJ8yiBieAC0UHBwACbN3CB+YgRk3gDCEObmYsjVZGRtkcQUAIIwfgQ6OPBycBI2DYkFgXQw5mQA6nQE8HieYHAmSkcEAG5HGsNgeNog1wEAklAtsCZgEgQCU0+JhgGgoG0KID40mce30fGGULgzgk1zsPCIyJg5TExEG4MCngkyOGOjEOjgYx4UKIMD2BCaMDoyAAHAAGHgSOAlAic7G4oZ5JI8GBkukMplIABMPHwcAAKmTlPkIXBzOSmi0mlYxCQcDi4AAxCASQazbzDMSLCAgdjsIA)

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Errors.

### Actual Result

Successfully parsed.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `6.5.0` |

**Comment by bradzacher** [comment]
Note that we don't disallow `let let = value;` either.
I don't know if we (or prettier) really should care about whether or not an identifier is a valid identifier based on whether or not it's a disallowed pseudo-keyword.

I understand that babel disallows it - but IDK if it's worth us building the extra logic in for this case. It's not like we're making the AST more correct/safer - it's just a pedantic check.

**Comment by fisker** [comment]
I agree Prettier doesn't really care about this either. But I think it worth to raise an issue here, we expect only valid code be parsed after all.

**Comment by JoshuaKGoldberg** [comment]
It _seems_ to me that this can be closed as a wontfix - is that a correct interpretation?

**Comment by bradzacher** [comment]
Yeah we can close this out.
From our perspective this isn't syntactically wrong - it is semantically (which is TS's perspective).
